### PR TITLE
Build updates including checking for already published artifacts BT-250

### PR DESCRIPTION
**Pull request by kshakir**

### CODE REVIEW

**Comment by kshakir** [comment]
Example error output here: https://travis-ci.com/github/broadinstitute/cromwell/jobs/516477139

File: project/Publishing.scala [hunk 1]
```diff
@@ -97,7 +101,7 @@ object Publishing {
       cache = false,
       removeIntermediateContainers = BuildOptions.Remove.Always
     ),
-    ThisBuild / dockerCustomSettings := Nil, // setting the default value
+    ThisBuild / dockerCustomSettings := Nil // setting the default value
```
Comment (mcovarr) [inline project/Publishing.scala:104]: TOL dangling commas are an error now? ugh...

File: project/Publishing.scala [hunk 2]
```diff
@@ -151,14 +160,94 @@ object Publishing {
...
-    publishTo := Option(artifactoryResolver(isSnapshot.value)),
+  /**
+    * The types of MDArtifacts published by this sbt build.
+    *
+    * This static list is an alternative to retrieving the types from sbt's `publishConfiguration` and its `artifacts`.
+    * That `publishConfiguration` unfortunately depends on `compile` so it takes several minutes to run.
+    */
+  private val cromwellMDArtifactTypes = List(
+    CromwellMDArtifactType("pom", "pom", None),
+    CromwellMDArtifactType("jar", "jar", None),
```
Comment (mcovarr) [inline project/Publishing.scala:179]: TOL pom poms and jar jar in the same PR...

**Review by mcovarr — COMMENTED** [review]
definitely going to want a walkthrough... 🙂

**Review by breilly2 — APPROVED** [review]

File: project/Publishing.scala (hunk 1 above)
Comment (kshakir) [reply project/Publishing.scala:104]: After 3+ years, I've given up on IntelliJ's scala plugin fixing its linting warnings for `project/*.scala` files. 😞

Trailing commas are valid in SBT, and preferred for source-controlled list items. But for developers, the false errors in the linter mask other issues.

**Review by mcovarr — APPROVED** [review]

**Pull request merged**